### PR TITLE
less dupes, cleanup

### PR DESCRIPTION
**Pull request by TriplEight**
- less dupes:
  `&name` is called "the yaml anchor", and used to reduce dupes, like that:
```
jobname:          &name
  stage:          test
  variables:
    var1:         1
  script:
    - do that

another-jobname:
  # it is the same as "jobname`, but the script is different
  <<:   *name
  script:
    - do this
```

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -286,25 +287,23 @@ build-linux-substrate:
...
-    - printf '\n# building node-template\n\n'
-    - ./.maintain/node-template-release.sh ./artifacts/substrate/substrate-node-template.tar.gz
-    - cp -r .maintain/docker/substrate.Dockerfile ./artifacts/substrate/
     - sccache -s
 
+build-linux-node-template:
+  <<:                              *build-binary
+  script:
+      - ./.maintain/node-template-release.sh ./artifacts/substrate/substrate-node-template.tar.gz
+      - cp -r .maintain/docker/substrate.Dockerfile ./artifacts/substrate/
```
Comment (bkchr) [inline .gitlab-ci.yml]: Why do we do this here? And not in the build substrate job above?

Comment (TriplEight) [reply .gitlab-ci.yml]: you're right, right now it's not effective to split these jobs

Comment (TriplEight) [reply .gitlab-ci.yml]: actually, we are compiling it right there: https://gitlab.parity.io/parity/substrate/-/jobs/339822#L706 , during `cargo build`.
Can we use that artifact?

Comment (bkchr) [reply .gitlab-ci.yml]: I was speaking about the docker file. Putting the node template into its own job sounds reasonable.

**Pull request merged**